fastplotlib/graphics/_base.py

```diff
@@ -68,3 +71,72 @@ def __repr__(self):
         else:
             return f"fastplotlib.{self.__class__.__name__} @ {hex(id(self))}"
 
+class Interaction(ABC):
+    @property
+    def indices(self) -> Any:
+        return self.indices
+
+    @indices.setter
+    def indices(self, indices: Any):
+        self.indices = indices
+
+    @property
+    @abstractmethod
+    def features(self) -> List[str]:
+        pass
+
+    @abstractmethod
+    def _set_feature(self, feature: str, new_data: Any, indices: Any):
+        pass
+
+    @abstractmethod
+    def _reset_feature(self, feature: str, old_data: Any):
+        pass
+
+    def link(self, event_type: str, target: Any, feature: str, new_data: Any, indices_mapper: callable = None):
+        valid_events = ["click"]
+        if event_type in valid_events:
+            self.world_object.add_event_handler(self.event_handler, event_type)
+        else:
+            raise ValueError("event not possible")
+
+        if isinstance(target.data, List):
+            old_data = list()
+            for line in target.data:
+                old_data.append(getattr(line, feature).copy())
+        else:
+            old_data = getattr(target, feature).copy()
+
+        if event_type in self.registered_callbacks.keys():
+            self.registered_callbacks[event_type].append(
+                CallbackData(target=target, feature=feature, new_data=new_data, old_data=old_data, indices_mapper=indices_mapper))
+        else:
+            self.registered_callbacks[event_type] = list()
+            self.registered_callbacks[event_type].append(
+                CallbackData(target=target, feature=feature, new_data=new_data, old_data=old_data, indices_mapper=indices_mapper))
+
+    def event_handler(self, event):
+        if event.type == "click":
+            # storing click information for each click in self.indices
+            #self.indices(np.array(event.pick_info["index"]))
+            click_info = np.array(event.pick_info["index"])
+        if event.type in self.registered_callbacks.keys():
+            for target_info in self.registered_callbacks[event.type]:
+                # need to map the indices to the target using indices_mapper
+                if target_info.indices_mapper is not None:
+                    indices = target_info.indices_mapper(target=target_info.target, indices=click_info)
```

Comment:
an `indices_mapper` function should be given source and target, something like this:

```
indices_mapper(source=self, target=target_info.target, event_indices=click_info
```

And the user can write their function to take in all 3 and do whatever they want.

Reply:
yes, I assumed that some general form would need to be established